### DISCUSSION TITLE
Some questions about Brinkman source term & Immersed boundary method

### DISCUSSION THREAD

**Discussion by kanjunao** [ExtremeFLOW/neko#1305] [General]
Hi guys
I'm very happy to communicate with you! 
I have been trying to use high-order methods to simulate a deformation problem. Currently I am learning about the Brinkman source term in neko. I think this source term is similar to the immersed boundary force in IBM. They all simulate solid surfaces through artificial assignment of values, which is setting the velocity inside the solid and on the surface to 0. This is a great way to simulate complex interfaces.

But when we talk about deformation, for example, for an Euler-Bernoulli beam (see in the pic), we also need to give the corresponding equation of motion of the solid. That is, fluid interacts with solid (making the beam bend), and solid also has an effect on the fluid part. We can understand this as a two-way coupling between fluid and solid. Combined with neko, theoretically we should add motion equations to Point zones so that they can determine their motion (bending) based on the stress on the surface? 
![pic_bend](https://github.com/ExtremeFLOW/neko/assets/45754632/f539fd21-f82d-4803-8ec6-62fd05f7d37a)

I would like to know what suggestions and opinions you have? Can neko simulate this type of problem? :)
Sincerely
Kan

**Comment by timfelle** [comment]
Hi @kanjunao 

This is indeed a very interesting case you are bringing up.
It is closely related to the topology optimization some of us are also working with, here we assume stiff structures, but need to update the shape of them over time.

It is indeed an interesting proposal but currently not supported as you rightly suspected.
In terms of implementation, the forcing is applied as a field defined everywhere, but with 0 values outside structures, and some user defined value iside the structure. This mean you can in a user_check function update the values of the field as you see fit. The brinkman term is not inherently tied to a point zone at the moment, but only use it to define the forcing field. However, you touch on a great point that it might be more useful to tie it directly to the point zones, and then extend point zones to be more general than the simple shapes they represent now. However, that would take quite some development.

The code snippet bellow describe how one could get access to the brinkman field in a user_check.

```fortran
use neko
subroutine modify_brinkman(....)
!...
type(field_t), pointer :: brinkman

brinkman = neko_field_registry%get_field('brinkman')
```

**Reply by timfelle** [reply]
My comment that it would take ome development should be read as "It will take some time before we have official support for this", rather than "We are not going to spend time doing that." 😅
